Fixed some bugs and updated reverse_model

egesko opened the pull request:
- Fixed HistogramVAEWrapper not working with `wrapped_model.evaluate()`, and therefore also with `wrapped_model.fit()`.
- Moved reverse_model and reverse_layer to capsa.utils
- Added Flatten layer compatibility to reverse_model and reverse_layer functions. Now, if layer to be reversed is Flatten, we change it to Reshape with the output shape of previous layer in Encoder.

egesko commented:
I also like to remind that encoder-decoder conversion is not always perfect. As in, input_shape of encoder does not always match the output_shape of decoder, even when the Conv-ConvTranspose layers have the same configs. It is not currently possible to estimate when this works, and when it doesn't.

aamini commented:
Thanks @egesko! To overcome the issue of output shapes not matching inputs do you think this approach would fix it:
After every layer, what if we resized that layers output to the same size as the corresponding layer in the encoder. The resizing operation is commonly done via bilinear interpolation layers in this case. 

What do you think?

egesko commented:
- Updated the `reverse_model` to include Resizing layer with billinear configuration
- Updated `reverse_layer` to add both Reshape+Dense layer instead of only Reshape layer.
- Solved a merge conflict at `capsa/utils/init.py`.
